src/runtime/io.cpp

```diff
@@ -630,6 +631,165 @@ extern "C" LEAN_EXPORT obj_res lean_io_prim_handle_put_str(b_obj_arg h, b_obj_ar
+        ucal_close(cal);
+        return io_result_mk_ok(mk_option_none());
+    }
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get next transation")));
+    }
+
+    int32_t tzIDLength = ucal_getTimeZoneDisplayName(cal, is_dst ? UCAL_DST : UCAL_STANDARD, "en_US", tzID, 32, &status);
+    
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to timezone identifier")));
+    }
+
+    char dst_name[256];
+    u_strToUTF8(dst_name, sizeof(dst_name), NULL, tzID, tzIDLength, &status);
+
+    UChar display_name[32];
+    int32_t display_name_len = ucal_getTimeZoneDisplayName(cal, is_dst ? UCAL_SHORT_DST : UCAL_SHORT_STANDARD, "en_US", display_name, 32, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to read abbreaviation")));
+    }
+
+    char display_name_str[256];
+    u_strToUTF8(display_name_str, sizeof(display_name_str), NULL, display_name, display_name_len, &status);
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get abbreviation to cstr")));
+    }
+
+    int32_t zone_offset = ucal_get(cal, UCAL_ZONE_OFFSET, &status);
+    zone_offset += dst_offset;
+
+    if (U_FAILURE(status)) {
+        ucal_close(cal);
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get zone_offset")));
+    }
+
+    int64_t tm = (int64_t)(nextTransition / 1000.0);
+    ucal_close(cal);
+
+    int offset_seconds = zone_offset / 1000;
+
+    lean_object *lean_tz = lean_alloc_ctor(0, 3, 1);
+    lean_ctor_set(lean_tz, 0, lean_int_to_int(offset_seconds));
+    lean_ctor_set(lean_tz, 1, lean_mk_string_from_bytes_unchecked(dst_name));
+    lean_ctor_set(lean_tz, 2, lean_mk_string_from_bytes_unchecked(display_name_str));
+    lean_ctor_set_uint8(lean_tz, sizeof(void*)*3, is_dst);
+    
+    lean_object *lean_pair = lean_alloc_ctor(0, 2, 0);
+    lean_ctor_set(lean_pair, 0, lean_int64_to_int(tm));
+    lean_ctor_set(lean_pair, 1, lean_tz);
+
+    return lean_io_result_mk_ok(mk_option_some(lean_pair));
+#else
+    return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to get timezone, its windows only.")));
+#endif
+}
+
+/* Std.Time.Database.Windows.getLocalTimeZoneIdentifierAt : @&Int64 → IO String */
+extern "C" LEAN_EXPORT obj_res lean_get_windows_local_timezone_id_at(uint64_t tm_obj, obj_arg /* w */) {
+#if defined(LEAN_WINDOWS)
+    UErrorCode status = U_ZERO_ERROR;
+    UCalendar* cal = ucal_open(NULL, -1, NULL, UCAL_GREGORIAN, &status);
+
+    if (U_FAILURE(status)) {
+        return lean_io_result_mk_error(lean_decode_io_error(EINVAL, mk_string("failed to open calendar")));
+    }
+0
```

Comment:
Oops, this was a good reminder to put the `release-ci` label so that we actually try to compile the Windows code on CI.